lib/buffer.js

```diff
@@ -62,20 +68,18 @@ Buffer.prototype.swap32 = function swap32() {
 const flags = bindingObj.flags;
 const kNoZeroFill = 0;
 
-function createBuffer(size, noZeroFill) {
-  flags[kNoZeroFill] = noZeroFill ? 1 : 0;
+function createUnsafeBuffer(size) {
+  flags[kNoZeroFill] = 1;
   try {
-    const ui8 = new Uint8Array(size);
-    Object.setPrototypeOf(ui8, Buffer.prototype);
-    return ui8;
+    return new FastBuffer(size);
```

Comment:
See the line above - https://github.com/nodejs/node/pull/6893/files#diff-196d056a936b6d2649721eb639e0442bR10 - I'm assigning `constructor` to `Buffer`, so the `FastBuffer` is invisible to the consumer.